substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/ObjectScanner.java

```diff
@@ -68,6 +69,16 @@ public class ObjectScanner {
     private final CompletionExecutor executor;
     private final Deque<WorklistEntry> worklist;
     private final ObjectScanningObserver scanningObserver;
+    private Predicate<JavaConstant> shouldScanConstant = c -> true; // always true by default
+    private Predicate<AnalysisField> shouldScanField = c -> true;   // always true by default
```

Comment:
This predicates don't need to be on the base `ObjectScanner`. You can extend `ObjectScanner` and add the specific checks there, then you can make a `StandaloneHeapSnapshotVerifier` and install the custom `ObjectScanner` there.

Reply:
Fixed as suggested.